plugins/catalog-backend/src/database/operations/stitcher/buildEntitySearch.ts

```diff
@@ -133,6 +133,9 @@ export function mapToRows(input: Kv[], entityId: string): DbSearchRow[] {
 
   for (const { key: rawKey, value: rawValue } of input) {
     const key = rawKey.toLocaleLowerCase('en-US');
+    if (MAX_KEY_LENGTH < key.length) {
```

Comment:
Can we switch this around, it's hard to parse.

`if (key.length > MAX_KEY_LENGTH)` reads a little easier.

Reply:
Curiously, I wrote it this way exactly for the reason that you are asking me to change it.

I use only `<` and `<=` for comparisons because they match the natural sorting order. The smaller value on the left, the large value on the right, this is how we have been taught in school when we learned about [the number line](https://en.wikipedia.org/wiki/Number_line).

Using this rule, I write a condition like _"`x` between `a` and `b`"_ as `a < x && x < b` and this way, `x` is both logically and visually _between_ `a` and `b`. The negation of this condition becomes `x < a || b < x` and in my head it becomes _"x is outside the (a, b) range"_.

I don't want to convince you of anything, I just explained how I write conditions and how I teach others to write them, to be readable and visually self-explanatory.

I also do not have any argument against your suggestion. If it seems easier for you, here you are: I updated the PR.

Reply:
Interesting take. I see where you're coming from but hard agree with @benjdlambert and I suspect the majority of people would too: `if (key.length > MAX_KEY_LENGTH)` reads like English and perfectly expresses _intent_ and the condition being violated, and lends itself to linguistically reading as "x _exceeds_ y which is a bad thing". Which in my mind trumps most other concerns. The risk for people of all walks of life - mathematically inclined or not, visual thinkers or not, senior devs or not - of misreading one more often than the other is likely larger.

Plus one more thing. Your code actually had a bug, maybe in part because of the order. In what we feel is the natural order, you also in a natural way pick between `>` and `>=` based on whether the constant was meant to be inclusive or not. Note that we changed from `<` to `>` - not `>=` which otherwise would have been the case.

Reply:
> Plus one more thing. Your code actually had a bug, maybe in part because of the order. In what we feel is the natural order, you also in a natural way pick between > and >= based on whether the constant was meant to be inclusive or not. Note that we changed from < to > - not >= which otherwise would have been the case.

Nope. After @benjdlambert's suggestion, I changed [from `a < b` to `b > a`](https://github.com/backstage/backstage/pull/23337/commits/fe56db540042466b75e4e760e6348d2074c694ef#diff-f89854e1c2ce5a5879141a848d56d02b434a7e73711fda2342a84f22ab95060aL136), which is the same thing. If it had a bug then it still has the same bug. Or both versions are correct.

The code on `master` reads `if (key.length <= MAX_KEY_LENGTH) { ... process this item ... }`, I changed it to `if (key.length > MAX_KEY_LENGTH) { ... skip this item ... }`. Where is the bug?

<img width="1103" alt="image" src="https://github.com/backstage/backstage/assets/1086598/94d18379-1609-4fa7-8481-40bc3cadd5ac">

Reply:
Hah, see, I even confused myself!